### PR TITLE
feat: Add data proxy docs

### PR DESCRIPTION
**Pull request by albertoperdomo**
This PR adds the data proxy to our Prisma Data Platform docs.

To review:
* Preview deployment is at https://deploy-preview-2407--prisma2-docs.netlify.app/
* Navigate to Concepts -> Components -> Prisma Data Platform

### CODE REVIEW

**Review by br0kend — APPROVED** [review]
lgtm

File: content/200-concepts/100-components/07-prisma-data-platform.mdx [hunk 1]
```diff
@@ -39,3 +39,158 @@ In order to add users, you need to use their GitHub username and they need to ha
...
+
+> The Data Proxy is currently available in the AWS Frankfurt (eu-central-1) and N. Virginia (us-east-1) regions. If you'd like support for other providers or regions, [we'd love to hear from you](https://forms.gle/17uyd1raniRtGSPe9).
+
+Copy the connection string displayed and store it securely, similarly to how you would treat a database credentials. You may generate more connection strings in this section if you wish and of course, revoke them if you don't need them anymore.
+
+Connection strings have the following format: `prisma://{DEPLOYMENT_LOCATION}.prisma-data.com/?api_key={DATA_PROXY_API_KEY}`
+
+#### Step 2: Install Data Proxy-enabled Prisma stack
+
+To use the data proxy, you will version `3.3.0` or higher of the `prisma` and `prisma-client` npm packages.
```
Comment (Jolg42) [inline content/200-concepts/100-components/07-prisma-data-platform.mdx]:
```suggestion
To use the data proxy, you will need the version `3.3.0` or higher of the `prisma` and `@prisma/client` npm packages. 
```

File: content/200-concepts/100-components/07-prisma-data-platform.mdx [hunk 2]
```diff
@@ -39,3 +39,158 @@ In order to add users, you need to use their GitHub username and they need to ha
...
+
+Because of this, using traditional databases in serverless functions often leads to exhausting the number of concurrent database connections and increased latencies on each request to establish a database connection.
+
+### Using the Data Proxy in a Prisma application
+
+We assume you already have an account and a project on [Prisma Cloud](https://cloud.prisma.io). If you don't please do that before continuing.
+
+#### Step 1: Generate a Data Proxy connection string
+
+Navigate to the "Data Proxy" tab of your Prisma Cloud project, select desired proxy location, and click on "Enable".
```
Comment (Jolg42) [inline content/200-concepts/100-components/07-prisma-data-platform.mdx]:
```suggestion
Navigate to the "Data Proxy" tab of your Prisma Cloud project, select the desired proxy location, and click on "Enable".
```

File: content/200-concepts/100-components/07-prisma-data-platform.mdx [hunk 3]
```diff
@@ -39,3 +39,158 @@ In order to add users, you need to use their GitHub username and they need to ha
...
+
+To use the data proxy, you will version `3.3.0` or higher of the `prisma` and `prisma-client` npm packages. 
+
+```terminal
+npm install prisma --save-dev
+```
+
+#### Step 3: Enable the feature flag in the Prisma schema file
+
+In the generator block of your `Prisma.schema` file, enable the feature flag:
```
Comment (Jolg42) [inline content/200-concepts/100-components/07-prisma-data-platform.mdx]:
```suggestion
In the generator block of your `schema.prisma` file, enable the feature flag:
```

File: content/200-concepts/100-components/07-prisma-data-platform.mdx [hunk 4]
```diff
@@ -39,3 +39,158 @@ In order to add users, you need to use their GitHub username and they need to ha
...
+To generate the client for the deployment, a specific environment variable `PRISMA_CLIENT_ENGINE_TYPE="dataproxy"` needs to be set:
+
+```terminal
+PRISMA_CLIENT_ENGINE_TYPE='dataproxy' npx prisma generate
+```
+
+#### Running the app
+
+In the deployment you want to use the proxy, make sure that the `DATABASE_URL` environment variable is set to the proxy connection string. 
+Currently, the only supported datasource name is `db`. Make sure your `prisma.schema` file includes the `datasource db { ... }` block.
```
Comment (Jolg42) [inline content/200-concepts/100-components/07-prisma-data-platform.mdx]:
```suggestion
Currently, the only supported datasource name is `db`. Make sure your `schema.prisma` file includes the `datasource db { ... }` block.
```

File: content/200-concepts/100-components/07-prisma-data-platform.mdx [hunk 5]
```diff
@@ -39,3 +39,158 @@ In order to add users, you need to use their GitHub username and they need to ha
...
+
+To generate the client for the deployment, a specific environment variable `PRISMA_CLIENT_ENGINE_TYPE="dataproxy"` needs to be set:
+
+```terminal
+PRISMA_CLIENT_ENGINE_TYPE='dataproxy' npx prisma generate
+```
+
+#### Running the app
+
+In the deployment you want to use the proxy, make sure that the `DATABASE_URL` environment variable is set to the proxy connection string.
```
Comment (Jolg42) [inline content/200-concepts/100-components/07-prisma-data-platform.mdx]:
```suggestion
In the deployment where you want to use the Data Proxy, make sure that the `DATABASE_URL` environment variable is set to the Data Proxy connection string. 
```

File: content/200-concepts/100-components/07-prisma-data-platform.mdx [hunk 6]
```diff
@@ -39,3 +39,158 @@ In order to add users, you need to use their GitHub username and they need to ha
...
+```prisma
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+```
+
+#### Step 4: Start your app
+
+We don't know how you run your app, but if you use `env` in your schema, nothing but the value of your environment variable should change. For example:
```
Comment (Jolg42) [inline content/200-concepts/100-components/07-prisma-data-platform.mdx]:
```suggestion
We don't know how you run your app, but if you use `env()` in your schema, nothing but the value of your environment variable should change. For example:
```

File: content/200-concepts/100-components/07-prisma-data-platform.mdx [hunk 7]
```diff
@@ -39,3 +39,158 @@ In order to add users, you need to use their GitHub username and they need to ha
...
+
+#### Step 1: Generate a Data Proxy connection string
+
+Navigate to the "Data Proxy" tab of your Prisma Cloud project, select desired proxy location, and click on "Enable".
+
+![Data Proxy tab](./data-proxy-tab.png)
+
+> The Data Proxy is currently available in the AWS Frankfurt (eu-central-1) and N. Virginia (us-east-1) regions. If you'd like support for other providers or regions, [we'd love to hear from you](https://forms.gle/17uyd1raniRtGSPe9).
+
+Copy the connection string displayed and store it securely, similarly to how you would treat a database credentials. You may generate more connection strings in this section if you wish and of course, revoke them if you don't need them anymore.
```
Comment (millsp) [inline content/200-concepts/100-components/07-prisma-data-platform.mdx]:
```suggestion
Copy the connection string displayed and store it securely, similarly to how you would treat database credentials. You may generate more connection strings in this section if you wish and of course, revoke them if you don't need them anymore.
```

File: content/200-concepts/100-components/07-prisma-data-platform.mdx [hunk 8]
```diff
@@ -39,3 +39,158 @@ In order to add users, you need to use their GitHub username and they need to ha
...
+Setting a concurrency limit on AWS Lambda should not be required, but you should be aware that requests will be queued if necessary until they can be processed.
+
+#### Proxy Limits
+
+The limits for running different schemas concurrently during Early Access are:
+
+- Up to 3 schemas can be used at one moment in a single Prisma app (for example: while updating your app to a new schema version)
+- Up to 5 schemas can be used at one moment account-wide
+
+The Data Proxy is architected to seamlessly scale to support large workloads, but during the Early Access period there will be a fairly low scale limit. The actual performance for a single schema using the data proxy will depend on a large number of variables (the type of queries, latencies and deployment locations, the performance of your database, ...). The data proxy will not yield more 200 rps. The actual number is hard to predict is likely to be well below.
```
Comment (Jolg42) [inline content/200-concepts/100-components/07-prisma-data-platform.mdx]:
```suggestion
The Data Proxy is architected to seamlessly scale to support large workloads, but during the Early Access period there will be a fairly low scale limit. The actual performance for a single schema using the data proxy will depend on a large number of variables (the type of queries, latencies and deployment locations, the performance of your database, ...). The data proxy will not yield more than 200 requests per second. The actual number is hard to predict is likely to be well below.
```

File: content/200-concepts/100-components/07-prisma-data-platform.mdx [hunk 9]
```diff
@@ -39,3 +39,158 @@ In order to add users, you need to use their GitHub username and they need to ha
...
+
+Please reach out to us if you're seeing `429` HTTP errors in your logs - this may mean you're hitting the quotas - we'd love to know about your use case and work with you on this.
+
+#### Cost of the service
+
+The Data Proxy is provided for free during Early Access. We will use this period to learn about usage and tweak the pricing model. When the Prisma Data Platform is launched in General Availability, there will be continue to be an option to use the Data Proxy for free with some limitations, alongside paid options.
```
Comment (Jolg42) [inline content/200-concepts/100-components/07-prisma-data-platform.mdx]:
```suggestion
The Data Proxy is provided for free during Early Access. We will use this period to learn about usage and tweak the pricing model. When the Prisma Data Platform is launched in General Availability, there will an option to use the Data Proxy for free with some limitations, alongside paid options.
```

File: content/200-concepts/100-components/07-prisma-data-platform.mdx [hunk 10]
```diff
@@ -39,3 +39,158 @@ In order to add users, you need to use their GitHub username and they need to ha
...
+```
+
+#### Step 3: Enable the feature flag in the Prisma schema file
+
+In the generator block of your `Prisma.schema` file, enable the feature flag:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["dataProxy"]
```
Comment (millsp) [inline content/200-concepts/100-components/07-prisma-data-platform.mdx:97]: The casing can be anything, although that is what the engines autocomplete.

File: content/200-concepts/100-components/07-prisma-data-platform.mdx (hunk 5 above)
Comment (millsp) [inline content/200-concepts/100-components/07-prisma-data-platform.mdx]:
```suggestion
In the deployment where you want to use the proxy, make sure that the `DATABASE_URL` environment variable is set to the proxy connection string. 
```

File: content/200-concepts/100-components/07-prisma-data-platform.mdx (hunk 8 above)
Comment (millsp) [inline content/200-concepts/100-components/07-prisma-data-platform.mdx]:
```suggestion
The Data Proxy is architected to seamlessly scale to support large workloads, but during the Early Access period there will be a fairly low scale limit. The actual performance for a single schema using the data proxy will depend on a large number of variables (the type of queries, latencies and deployment locations, the performance of your database, ...). The data proxy will not yield more than 200 requests per second. The actual number is hard to predict is likely to be well below.
```

File: content/200-concepts/100-components/07-prisma-data-platform.mdx (hunk 9 above)
Comment (millsp) [inline content/200-concepts/100-components/07-prisma-data-platform.mdx]:
```suggestion
The Data Proxy is provided for free during Early Access. We will use this period to learn about usage and tweak the pricing model. When the Prisma Data Platform is launched in General Availability, there will be an option to continue using the Data Proxy for free with some limitations, alongside paid options.
```

**Review by millsp — CHANGES_REQUESTED** [review]

File: content/200-concepts/100-components/07-prisma-data-platform.mdx (hunk 7 above)
Comment (albertoperdomo) [reply content/200-concepts/100-components/07-prisma-data-platform.mdx]: it's either singular or plural.

* "a database credential" (without s) or 
* database credentials (in general)

Do you think plural sounds weird? I don't have a strong opinion

File: content/200-concepts/100-components/07-prisma-data-platform.mdx (hunk 10 above)
Comment (albertoperdomo) [reply content/200-concepts/100-components/07-prisma-data-platform.mdx:97]: If this is what automcomplete uses, let's stick with this. 😢

File: content/200-concepts/100-components/07-prisma-data-platform.mdx (hunk 7 above)
Comment (millsp) [reply content/200-concepts/100-components/07-prisma-data-platform.mdx]: I think that both are good

File: content/200-concepts/100-components/07-prisma-data-platform.mdx [hunk 11]
```diff
@@ -39,3 +40,163 @@ In order to add users, you need to use their GitHub username and they need to ha
...
 
 You can [create an issue](https://github.com/prisma/studio/issues/new?assignees=&labels=topic%3A+hosted+data+browser&template=hosted-data-browser-bug-report.md&title=) with your request and we will enable it for you. Please note that while this is a feature that will be part of a paid plan in the future, it is currently offered for free while we are in Early Access.
+
+## Data Proxy
+
+<Admonition type="warning">
+
+**Early Access Feature** <br />
+Please note that the Prisma Data Proxy is an early-stage product and should not be used in production environments?
+</Admonition>
```
Comment (tanberry) [inline content/200-concepts/100-components/07-prisma-data-platform.mdx:50]: @albertoperdomo did you mean to have a question mark at end of the Note?

**Comment by albertoperdomo** [comment]
No :(

File: content/200-concepts/100-components/07-prisma-data-platform.mdx (hunk 11 above)
Comment (albertoperdomo) [reply content/200-concepts/100-components/07-prisma-data-platform.mdx:50]: No :(
If this is live we should fix it.

**Pull request merged**